hazelcast/src/main/java/com/hazelcast/cp/internal/RaftService.java

```diff
@@ -636,8 +637,14 @@ void updateMissingMembers() {
                     logger.warning(cpMember + " is not present in the cluster. It will be auto-removed after "
                             + config.getMissingCPMemberAutoRemovalSeconds() + " seconds.");
                 }
-            } else if (missingMembers.remove(cpMember) != null) {
-                logger.info(cpMember + " is removed from the missing members list as it is in the cluster.");
+            } else if (missingMembers.containsKey(cpMember)) {
+                OperationServiceImpl operationService = nodeEngine.getOperationService();
```

Comment:
Is it ok to do network calls here? I can see call paths of `updateMissingMembers` from raft operations (eg `RemoveCPMemberOp`, `InitMetadataRaftGroupOp` etc), so blocking the operation thread for network access is not a good idea. Can you consider alternatives to have the CP member UUID already available at this point?

Reply:
Yes, I'll take a look at other places we can potentially use to get the actual CP member UUID

Reply:
Have changed the blocking operation to async to prevent the operation thread blocking.

Regarding other places where we can potentially get CP member UUID:
- _Member_ attribute - when a node joins to cluster the CP UUID could be stored as a _Member_ attribute, but unfortunately, attributes should be set before the instance has been started.
- nodes _AppendRequest_ response contains the member CP UUID - this CP UUID will be known only by the Raft-group Leader (and unavailable on Followers). But `missingMembers` list is tracked on every CP node, so the leader should in some way notify CP Members to update their `missingMembers` list.

Regarding other Raft operations (eg _RemoveCPMemberOp_, _InitMetadataRaftGroupOp_ etc) - they invoked to change the CP group membership and update or get Metadata info. In our case, we want to get the CP UUID, if any, based on the node's IP address.